pkg/gui/context/base_context.go

```diff
@@ -197,6 +200,10 @@ func (self *BaseContext) NeedsRerenderOnWidthChange() bool {
 	return self.needsRerenderOnWidthChange
 }
 
+func (self *BaseContext) NeedsRerenderOnHeightChange() bool {
+	return self.needsRerenderOnHeightChange
+}
+
```

Comment:
It's a bit unfortunate that this is a separate flag that clients must remember to set when they set renderOnlyVisibleLines on the ListContextTrait. I guess this could have been avoided by putting it on `Context` instead of `IBaseContext`, and then implementing it in `ListContextTrait` to return renderOnlyVisibleLines. I haven't tried this yet, let me know if you think it's worth trying.

Reply:
That sounds reasonable to me

Reply:
I tried, but it doesn't seem to be possible with the way ListContextTrait is implemented, since ListContextTrait embeds a context (usually a SimpleContext). It looks like we would first have to do something about the containment hierarchy of contexts, and this seems like too big of a change for this PR. Plus, I wouldn't actually know how. 😅

Reply:
Yeah the context hierarchy is indeed a pickle. Happy to leave for another PR